### PR TITLE
feat(pb_input_wc): New pb-input WC

### PR DESCRIPTION
**Pull request by jeromecambon**
- First version of pb-input web component
- Limitation: translation to do

Covers [UID-358](https://bonitasoft.atlassian.net/browse/UID-358)

### CODE REVIEW

File: components/packages/pb-input/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "@bonitasoft/pb-input",
+  "version": "1.0.5",
+  "description": "A simple search box",
```
Comment (jbaudot85) [inline components/packages/pb-input/package.json:4]: Is it really a search box ? ;)

File: components/packages/pb-input/src/pb-input.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  private readOnly: boolean = false;
+
+  @property({ attribute: 'label-hidden', type: Boolean, reflect: true })
+  private labelHidden: boolean = false;
+
+  @property({ attribute: 'label', type: String, reflect: true })
+  private label: string = "Default label";
+
+  @property({ attribute: 'label-position', type: String, reflect: true })
+  private labelPosition: string = "top";
```
Comment (benjaminParisel) [inline components/packages/pb-input/src/pb-input.ts:55]: When i change this value, nothing occurs

File: components/packages/pb-input/src/pb-input.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  private type: string = "text";
+
+  @property({ attribute: 'min', type: String, reflect: true })
+  private min: string = "";
+
+  @property({ attribute: 'max', type: String, reflect: true })
+  private max: string = "";
+
+  @property({ attribute: 'step', type: String, reflect: true })
+  private step: string = "1";
```
Comment (benjaminParisel) [inline components/packages/pb-input/src/pb-input.ts:76]: If i put value="10" and step="10", on my first click on increase value, value is 11 and not 10. Second increase are good (11 => 21, 21 => 31)

File: components/packages/pb-input/src/pb-input.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      
+      .label-elem {
+        font-size: 14px;
+        font-weight: 700;
+        margin-bottom: 5px;
+      }
+      
+      /* Add a red star after required inputs */
+      .label-required:after {
+        content: "*";
```
Comment (benjaminParisel) [inline components/packages/pb-input/src/pb-input.ts:135]: put space before * to be more confortable display

File: components/packages/pb-input/src/pb-input.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        <label
+          class="${this.getLabelCssClass()}"
+          for="input"
+        >${this.label}</label>
+        `
+  }
+
+  private getLabelCssClass() : string {
+    return "control-label label-elem " + (this.required ? "label-required" : "") +
+      " col-xs-" + (!this.labelHidden && this.labelPosition === 'left' ? this.labelWidth : 12);
```
Comment (benjaminParisel) [inline components/packages/pb-input/src/pb-input.ts:178]: I don't think we want to manage only col-xs size... What about md, lg or others ?

**Review by benjaminParisel — CHANGES_REQUESTED** [review]
Some comments:
* What do you think about add an id attribute, to allow us/user, to test more easier his page ?
* I think, message when input is required or invalid is missing

File: components/packages/pb-input/package.json (hunk 1 above)
Comment (jeromecambon) [reply components/packages/pb-input/package.json:4]: Good catch, thanks!

**Comment by jeromecambon** [comment]
> Some comments:
> 
> * What do you think about add an id attribute, to allow us/user, to test more easier his page ?

Yes, I had this in mind as well, but forgot to add it.

> * I think, message when input is required or invalid is missing
I'm wondering if it should be part of the component or not... Need to think about it.

File: components/packages/pb-input/src/pb-input.ts (hunk 2 above)
Comment (jeromecambon) [reply components/packages/pb-input/src/pb-input.ts:55]: Yes indeed, need to check this

File: components/packages/pb-input/src/pb-input.ts (hunk 3 above)
Comment (jeromecambon) [reply components/packages/pb-input/src/pb-input.ts:76]: Ha... I don't reproduce this. Did you use :
 <pb-input type="number" value="10" step="10"></pb-input>

File: components/packages/pb-input/src/pb-input.ts (hunk 4 above)
Comment (jeromecambon) [reply components/packages/pb-input/src/pb-input.ts:135]: yes, agree

File: components/packages/pb-input/src/pb-input.ts (hunk 5 above)
Comment (jeromecambon) [reply components/packages/pb-input/src/pb-input.ts:178]: I did was was done in the legacy pbInput template. To be discussed if needed.

File: components/packages/pb-input/src/pb-input.ts (hunk 3 above)
Comment (benjaminParisel) [reply components/packages/pb-input/src/pb-input.ts:76]: I use:
```
 <pb-input label="Pouet" placeholder="Type a value" type="number" value="10" min="1" max="100"
    min-length="3" max-length="10" label-position="left" label-width="5" step="10" required="true"></pb-input>
```

Comment (jeromecambon) [reply components/packages/pb-input/src/pb-input.ts:76]: I reproduced the issue with an `<input>`:
`<input type="number" value="10" min="1" max="100" step="10">`
This is related to the `min="1"` which have this side effect...

**Review by benjaminParisel — APPROVED** [review]
LGTM, just small comment

File: components/packages/pb-input/src/pb-input.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+      .input-elem {
+        font-size: 14px;
+        height: 20px;
+      }
+      
+      .label-elem {
+        font-size: 14px;
+        font-weight: 700;
+        padding-left: 0
```
[the remark this answers is not in the document]
Comment (jeromecambon) [reply components/packages/pb-input/src/pb-input.ts:132]: Yes indeed, it behaves like this in the legacy widget.

**Pull request merged**